### PR TITLE
test: reset cucumber timeouts to practical limits

### PR DESCRIPTION
**Pull request by hansieodendaal**
Description
---
Re-applied reasonable time outs to cucumber tests. It is not worthwhile
to have too stringent time outs as it makes the tests flaky on a slower
system and especially on CI.

Motivation and Context
---
Tests were flaky on slow and fast computers due to stringent time outs.

How Has This Been Tested?
---
Cucumber on slow and fast computers

### CODE REVIEW

**Comment by stringhandler** [comment]
Thanks for this, but it's going to be a nightmare if we keep going back and forward on these timeouts. 

I suggest we do this:
1. create constants 'SLOW','FAST' and set the timeouts in the steps to use them
2. on CI, we set 'SLOW' = 60 seconds, FAST to 10 seconds using a world parameter, but by default they are set to 20 seconds, 1 second
3. Remove all custom timeouts and only use SLOW, MEDIUM, FAST, 

Alternatively we can set it to `5  * seconds()` and use a speed factor in the world parameters

File: integration_tests/features/support/steps.js [hunk 1]
```diff
@@ -300,7 +300,7 @@ Given(
 
 Given(
   "I have {int} base nodes connected to all seed nodes",
-  { timeout: 20 * 1000 },
+  { timeout: 120 * 1000 },
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js]: this is too big a jump

File: integration_tests/features/support/steps.js [hunk 2]
```diff
@@ -2903,7 +2910,7 @@ Then(
 
 Then(
   /all wallets detect all transactions as Mined_Confirmed/,
-  { timeout: 40 * 1000 },
+  { timeout: 1200 * 1000 },
```
Comment (stringhandler) [inline integration_tests/features/support/steps.js:2927]: 20 minutes?

Comment (hansieodendaal) [reply integration_tests/features/support/steps.js:2927]: For stress tests

File: integration_tests/features/support/steps.js (hunk 1 above)
Comment (hansieodendaal) [reply integration_tests/features/support/steps.js]: Agreed, that one is not needed to be so high

**Comment by hansieodendaal** [comment]
> Thanks for this, but it's going to be a nightmare if we keep going back and forward on these timeouts.
> 
> I suggest we do this:
> 
> 1. create constants 'SLOW','FAST' and set the timeouts in the steps to use them
> 2. on CI, we set 'SLOW' = 60 seconds, FAST to 10 seconds using a world parameter, but by default they are set to 20 seconds, 1 second
> 3. Remove all custom timeouts and only use SLOW, MEDIUM, FAST,
> 
> Alternatively we can set it to `5 * seconds()` and use a speed factor in the world parameters

Let me think a bit about this and come up with a better solution than what we currently have. We have various value systems at play here; we do not want to wait a long time for a failing test, but we also do not want a test that will eventually succeed time out prematurely. This also behaves differently on slow and fast computers; the latter can many times be fixed with clever hold/synchronization points.

**Comment by stringhandler** [comment]
The CI passed using the previous values on https://github.com/tari-project/tari/pull/3494 and passed on 8 out of the 9 PR's following that (Github does not show circle ci results once it's merged, but I'm using the green and red ticks. They could have also failed on other checks. I'm excluding PRs to `validator-node` branch)

Given that 88% of the PR's passed, I would expect a far smaller adjustment to the timeouts

**Comment by stringhandler** [comment]
I think let's take this approach. As the tests fail, let's increase one or two steps at a time, as I have done in fad852051b0e7a1c8ac6c5930e0ac04bdf98228a

**Comment by hansieodendaal** [comment]
Ok, will make small PRs for the failing tests.

**Pull request closed without merging**